Learner wants the platform to remember that he/she is logged in for a long period of time, so that learner don't have to enter his/her details everytime.

snehasischatterjee opened the pull request:
This ticket is about the remembering users login for some days.
So the changes created are that once user logins and if he checks "Remember Login for 30 days" then user shouldn't logout of the portal until the cookie is alive. 
If he/she logouts intentionally then the cookie will get end and he will logged out to the Login page.

Some codes are changed to the moodle core files, Please use the patch to have the proper working.

snehasischatterjee committed:
Final commit for remembering user for 30 days

vijaysavakhande reviewed (approved):
Works fine at local environment.

hannahlillis commented:
Is it possible to remove the linting changes, as it's difficult to see which changes you've made, and which are just linting.

snehasischatterjee commented:
> Is it possible to remove the linting changes, as it's difficult to see which changes you've made, and which are just linting.

@hannahlillis 
All the green fields are my changes, however lot of them are added due to auto formatting. I can tell you the line numbers for my code which I have added.
**password_form.php = line no 54 to 58
index.php = line no 29 to 35
index.php = line no 130 to 143
lang/en/local_signin.php = line no 110**

I have also changed some code in the moodle core. And the patch is there in JIRA ticket.
https://avadolearning.atlassian.net/browse/ALP-677

hannahlillis commented on index.php:
```diff
@@ -123,7 +132,15 @@
+
+        if (isset($values->remembermedays)) {
+            global $USER;
+            $userDetailsArray = $USER->id;
```
We shouldn't need to use the variable type in the name e.g. $userDetails should be fine here

snehasischatterjee replied:
@hannahlillis 
Changes has been made. Please check & Review.

hannahlillis replied:
Is this actually an array or just an id? If it's just an int, it can just be called $userId, I think $userIdDetails is a little confusing

snehasischatterjee replied:
@hannahlillis 
Yupp changed to $userId.

snehasischatterjee committed:
Code changes for PR suggestions

snehasischatterjee committed:
PR suggestions

luisatheodoro reviewed (approved):
Looks good

luisatheodoro commented:
Applied patch and it's working fine.
